LM-3884 Quarkus Panache Repository pattern

Balazs-Joo opened the pull request.

Balazs-Joo committed:
LM-3884 Quarkus Panache Repository pattern

Balazs-Joo committed:
LM-3884 Quarkus Panache Repository pattern

Balazs-Joo committed:
LM-3884 Quarkus Panache Repository pattern

speter555 commented on qs-services/qs-panache-service/src/main/java/hu/icellmobilsoft/quarkus/sampler/panache/shutdown/ShutDownSupport.java:
```diff
@@ -0,0 +1,46 @@
+ * @author czenczl
+ * @since 0.1.0
+ */
+public class ShutDownSupport {
```
Én ezt valahova kiemelném mostmár commonba, tudom hogy itt is meg máshol is van helye, de ez nagyon core funkció lesz hosszútávon amire sok dolgot tudunk építeni.

speter555 commented on qs-services/qs-panache-service/pom.xml:
```diff
@@ -0,0 +1,210 @@
+            <version>3.9.2</version>
+        </dependency>
+<!-- dash not showing data due to missing instance name  -->
+<!--<dependency>-->
```
ez végső soron kell vagy nem?

czenczl replied:
ez bugos sztem még, ezért lett kiemelve külön docker compose-ba, és sztem azt is fogjuk használni a jövőben, könnyebb tesztelni és módosítani a dasheket

speter555 commented on qs-services/qs-panache-service/src/main/resources/import.sql:
szerintem pár példát nyugodtan beledobhatsz

speter555 reviewed (approved):
Nagyon szuper lett! egy dolgot hiányolok, a teszteket!

mate-biro commented on .github/workflows/compile_maven.yml:
```diff
@@ -15,7 +15,7 @@ jobs:
       matrix:
         os: [ 'ubuntu-latest']
-        java: [ '17' ]
+        java: [ '21' ]
```
itt is vannak java 17-ek:
- .github/workflows/maven-deploy.yml
- .github/workflows/manual-branch-RELEASE.yml
- .github/workflows/manual-main-RELEASE.yml

attila-kiss-it commented on qs-api/qs-api-dto/src/main/java/hu/icellmobilsoft/quarkus/sampler/api/jakarta/dto/exception/BaseRuntimeException.java:
```diff
@@ -0,0 +1,97 @@
+ * @author balazs.joo
+ * @since 0.1.0
+ */
+public class BaseRuntimeException extends RuntimeException {
```
Inkább a `coffee-se-api`-ba tennék ilyet, és `abstract class AbstractBaseRuntimeException`-ként, alapból két leszármazottal `BusinessRuntimeException` és `TechnicalRuntimeException`. Így meg lehetne szabadulni az eltúlzott `throws BaseException`-től "hiba esetén" szokástól, ki lenne kényszerítve legalább annyi, hogy üzleti vagy technikai hibáról van-e szó.

Balazs-Joo replied:
ez még a jövő zenéje :smile:

attila-kiss-it commented on qs-common/qs-common-model/src/main/java/hu/icellmobilsoft/quarkus/sampler/model/jpatest/SampleEntityProjection.java:
```diff
@@ -0,0 +1,33 @@
+ * limitations under the License.
+ * #L%
+ */
+package hu.icellmobilsoft.quarkus.sampler.model.jpatest;
```
A projection-öket másik package-be raknám, inkább a `model` mellé vagy alá, vagy még inkább a `query`-khez közel, mert azok definiálják az egyes projection-öket.

attila-kiss-it commented on qs-services/qs-panache-service/src/main/java/hu/icellmobilsoft/quarkus/sampler/panache/annotation/ParamName.java:
```diff
@@ -0,0 +1,63 @@
+ * Example usage:
+ * 
+ * <pre>{@code
+ * public void process(@ParamName("username") String name) {
```
"Ensure consistency to avoid confusion"
```suggestion
 * public void process(@ParamName("username") String username) {
```

attila-kiss-it commented on qs-services/qs-panache-service/src/main/java/hu/icellmobilsoft/quarkus/sampler/panache/annotation/ValidateIncomingParameters.java:
```diff
@@ -0,0 +1,71 @@
+@Inherited
+@InterceptorBinding
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ ElementType.METHOD, ElementType.TYPE })
```
A `Target` lehetne még `PARAMETER` is, így lehetőség nyílna nullable paraméterekre.

Balazs-Joo committed:
LM-3884 Quarkus Panache Repository pattern

attila-kiss-it commented:
🔥

czenczl commented on qs-services/qs-panache-service/src/main/java/hu/icellmobilsoft/quarkus/sampler/panache/annotation/TracedMethods.java:
```diff
@@ -0,0 +1,61 @@
+@InterceptorBinding
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ ElementType.METHOD, ElementType.TYPE })
+public @interface TracedMethods {
```
erre biztos szükségünk lesz? az otel lib csomagban vannak erre annotaciok előre készítve, nézzük majd meg

Balazs-Joo committed:
Merge branch 'main' into feature/LM-3884-quarkus-panache-active-record-pattern

Balazs-Joo committed:
LM-3884 Quarkus Panache Repository pattern

Balazs-Joo committed:
LM-3884 Quarkus Panache Repository pattern

Balazs-Joo committed:
Merge branch 'feature/LM-3884-quarkus-version-3.26.1' into feature/LM-3884-quarkus-panache-active-record-pattern

# Conflicts:
#	qs-common/qs-common-rest/src/main/java/hu/icellmobilsoft/quarkus/sampler/common/rest/action/BaseAction.java

Balazs-Joo committed:
LM-3884 Updated Panache MS

Balazs-Joo committed:
Merge branch 'main' into feature/LM-3884-quarkus-panache-active-record-pattern

Balazs-Joo committed:
LM-3884 Updated Panache MS

attila-kiss-it commented on qs-common/qs-common-core/src/main/java/hu/icellmobilsoft/quarkus/sampler/common/core/parameter/ValidateIncomingParametersInterceptor.java:
```diff
@@ -0,0 +1,149 @@
+            }
+            return ctx.proceed();
+        } catch (BaseException e) {
+            throw new BaseRuntimeException(e);
```
pl stream api-k használatakor nagyon jól jön a RTE

petrenyi-mark replied:
ez jogos, de interceptorban szerintem ez elég félrevezető, hogy megváltozik az exception.
van pl. egy metódus ami szignatúra szerint throws BaseException.
De ha rákerül ez az interceptor, akkor ez észrevehetelenül megváltoztatja és a hívó valójában már sose fog BaseException-t kapni, csak BaseRuntimeException-t. 
Egy validációs interceptornak nem szabadna, hogy ilyen mellékhatása legyen, ami a ctx.proceed-ből esik ki exception azon már semmiképp nem kellene változtatni.
Az ha validáció maga runtime-ot dob az már oké ugyanezen érvelés mentén, mert ott meg ha nem runtime akkor az a félrevezető, ha kimehet egy checked excpetion anélkül, hogy a metódus deklarálná

Balazs-Joo replied:
de amúgy ez így is van :) a wrappelés után van csak ctx.proceed(), de a try-ban nincs

Balazs-Joo committed:
Merge branch 'refs/heads/main' into feature/LM-3884-quarkus-panache-active-record-pattern

Balazs-Joo committed:
LM-3884 feat(panache): add MongoDB support with Panache entities and services

- Introduce new qs-common-mongodb module with MongoDB entities, repositories, and services
- Add REST APIs for MongoDB Panache operations (modify, query, projection, transaction)
- Update Docker Compose configurations for MongoDB setup
- Integrate MongoDB dependencies and configuration in panache service
- Add comprehensive test suites for MongoDB functionality

BREAKING CHANGE: Introduces new MongoDB dependencies and configuration requirements for the panache service.

Balazs-Joo committed:
LM-3884 refactor(mongodb): add JavaDoc and clean up MongoDB codec and entity classes

- Add comprehensive JavaDoc comments to MongoDB codec classes (MongoCustomCodecProducer, OffsetDateTimeCodec, OffsetDateTimeCodecProvider)
- Enhance AbstractMongoEntity with JavaDoc and getter/setter methods
- Refactor BasePanacheMongoService to use AbstractMongoEntity instead of MongoSampleEntity
- Reorganize imports and fix minor formatting issues across affected files
- Update test files with consistent newline endings

Closes LM-3884

Balazs-Joo committed:
LM-3884 refactor(panache-service): simplify mongo actions

remove unused endpoints and exception handling from mongo actions and drop the transaction helper dependency, keeping the service layer leaner

Balazs-Joo committed:
LM-3884 feat(common-core): enhance interceptors with conditional logging, validation, and empty collection support

- Refactor exception handling to include original exceptions in constructors and fix parameter logging
- Add conditional logging for non-public methods based on annotation attributes
- Introduce AllowEmptyCollection annotation for flexible parameter validation
- Update parameter validation to support non-public methods and empty collections
- Remove unused imports and simplify parameter name retrieval

BREAKING CHANGE: ValidateIncomingParameters annotation attribute renamed from 'validate' to 'validateNonPublicMethods

Balazs-Joo committed:
LM-3884 refactor(common-core): simplify interceptors

remove legacy non-public toggles from log and validation
annotations and rely solely on enabled attribute for both

almaxbacsi commented on qs-services/qs-panache-service/src/main/java/hu/icellmobilsoft/quarkus/sampler/panache/service/BasePanacheService.java:
```diff
@@ -0,0 +1,139 @@
+     * @return The entity if found, otherwise {@code null}.
+     */
+    public E findById(@ParamName("id") String id) {
+        return repository.findById(id);
```
Ez a `findById` nem talált rekord esetén null-t ad vissza, így a `HandleServiceExceptionsInterceptor` nem tudja a NoResultExceptiont BONotFound-ba forgatni. Ide jobb megoldás lenne:
`repository.find(AbstractIdentifiedAuditEntity_.ID, id).singleResult();` a singleResult metódus dobálja rendesen a NoResultExceptiont

Balazs-Joo committed:
LM-3884 review fix
